R/map.R

```diff
@@ -219,6 +222,29 @@ map_raw <- function(.x, .f, ...) {
   .Call(map_impl, environment(), ".x", ".f", "raw")
 }
 
+
+#' @rdname map
+#' @param .ptype If `NULL`, the default, the output type is the common type
+#'   of the elements of the result. Otherwise, supply a "prototype" giving
+#'   the desired type of output.
+#' @importFrom vctrs vec_c vec_size vec_ptype_common
+#' @export
+map_vec <- function(.x, .f, ..., .ptype = NULL) {
+  out <- map(.x, .f, ...)
+
+  .ptype <- vec_ptype_common(!!!out, .ptype = .ptype)
+  bad_sizes <- which(list_sizes(out) != 1L)
+  if (length(bad_sizes) >= 1) {
+    i <- bad_sizes[[1L]]
+    stop_bad_element_vector(out[[i]], i, .ptype, 1L, what = "Result")
+  }
```

Comment:
```suggestion
  bad_sizes <- list_sizes(out) != 1L
  if (any(bad_sizes)) {
    i <- which(bad_sizes)[[1L]]
    stop_bad_element_vector(out[[i]], i, .ptype, 1L, what = "Result")
  }
```

Probably a little faster